include/mqtt/endpoint.hpp

```diff
@@ -7426,16 +7422,33 @@ class endpoint : public std::enable_shared_from_this<endpoint<Socket, Mutex, Loc
     template <typename Iterator>
     typename std::enable_if<std::is_convertible<typename Iterator::value_type, char>::value>::type
     restore_serialized_message(packet_id_t /*packet_id*/, Iterator b, Iterator e) {
+        static_assert(
+            std::is_same<
+                typename std::iterator_traits<Iterator>::iterator_category,
+                std::random_access_iterator_tag
+            >::value,
+            "Iterator doesn't support random access"
```

Comment:
Maybe better wording:

Iterators provided to restore_serialized_message() must be random access iterators.

Reply:
Thanks, I use your wording.